src/shims/fs.rs

```diff
@@ -261,6 +261,85 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         this.try_unwrap_io_result(result)
     }
 
+    fn stat(
```

Comment:
Does this work only for macOS or also for Linux?
We should probably have an `assert!` for the target OS at the beginning of such functions to make sure they only get called on OSes they were meant for?

Reply:
only for macos, I'll add the assertion.